Implemented functionality to change route when on the mobile view to display the job configuration component(#24b1ket)

disha1202 opened the pull request:
### Related Issues
<!--  Put related issue number which this PR is closing. For example #123 -->

Closes #

### Short Description and Why It's Useful
<!-- Describe in a few words what is this Pull Request changing and why it's useful -->


### Screenshots of Visual Changes before/after (If There Are Any)
<!-- If you made any changes in the UI layer, please provide before/after screenshots -->


**IMPORTANT NOTICE** - Remember to add changelog entry


### Contribution and Currently Important Rules Acceptance
<!-- Please get familiar with following info -->

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

disha1202 committed:
Implemented functionality to change route when on the mobile view to display the job configuration component(#24b1ket)

disha1202 committed:
Added functionality to change route when on the mobile view to display the job configuration component(#24b1ket)

disha1202 committed:
Updated state to stre current job information(#24b1ket)

disha1202 committed:
Updated logic to store current job in state(#24b1ket)

disha1202 committed:
Added toast after skip, cancel and save actions and updated the markup(#24b1ket)

meet-aniket committed:
Merge branch 'main' of https://github.com/hotwax/job-manager into #24b1ket

meet-aniket committed:
Improved: code to make 'API' call when we do not have currentJob in state (#24b1ket).

meet-aniket committed:
Improved: code to make API call to fetch enumName for particlar job (#24b1ket).

meet-aniket committed:
Improved: code to add check if we do not have job related data for particular job (#24b1ket).

meet-aniket committed:
Improved: code to make component functional on refresh(#24b1ket).

meet-aniket committed:
Improved: code to update '.env.example' file (#24b1ket).

meet-aniket committed:
Merge branch 'main' of https://github.com/hotwax/job-manager into #24b1ket

meet-aniket committed:
Improved: code to add a watcher inside of setup method to update the option's title dynamically (#24b1ket).

meet-aniket committed:
Improved: the code to handle the case when we trying to use trigger in ion-modal but it does not work (#24b1ket).

dt2patel commented on src/components/JobConfiguration.vue:
```diff
@@ -4,7 +4,6 @@
       <h1>{{ $t(title) }}</h1>
       <ion-badge slot="end" color="dark" v-if="job?.runTime">{{ $t("running") }} {{ timeTillJob(job.runTime) }}</ion-badge>
     </ion-item>
-
```
unwanted change

meet-aniket replied:
Sir, reverted changes.

dt2patel commented on src/components/JobConfiguration.vue:
```diff
@@ -52,7 +51,6 @@
         <ion-input :placeholder="$t('occurrences')" v-model="count"/>
       </ion-item> -->
     </ion-list>
-
```
Unwanted change

meet-aniket replied:
Sir, reverted changes.

meet-aniket committed:
Improved: code to revert unwanted changes from 'JobConfiguration.vue' file (#24b1ket).

meet-aniket committed:
Merge branch 'main' of https://github.com/hotwax/job-manager into #24b1ket

meet-aniket committed:
Improved code to update .env.example file and variable as per this file as well (#24b1ket).

ymaheshwari1 commented on src/components/JobConfiguration.vue:
```diff
@@ -207,8 +214,10 @@ export default defineComponent({
             text: this.$t('Save'),
-            handler: () => {
-              this.updateJob();
+            handler:  () => {
```
```suggestion
            handler: () => {
```

meet-aniket replied:
Updated Sir.

meet-aniket committed:
Improved: code to revert unwanted changes from .env.example file (#24b1ket).

meet-aniket commented on src/components/JobConfiguration.vue:
```diff
@@ -15,13 +14,13 @@
         now commented the button and added a fallback string -->
         <!-- <ion-button id="open-run-time-modal" size="small" fill="outline" color="medium" v-show="!job?.runTime">{{ $t("Select run time") }}</ion-button> -->
-        <ion-modal trigger="open-run-time-modal">
+        <ion-modal :is-open="isModalOpen" @didDismiss="setOpen(false)">
```
Yes Sir.

meet-aniket committed:
Improved: code to revert unwanted space from handler function (#24b1ket).

adityasharma7 commented on src/components/JobConfiguration.vue:
```diff
@@ -10,13 +10,13 @@
         <ion-icon slot="start" :icon="timeOutline" />
         <ion-label>{{ $t("Run time") }}</ion-label>
-        <ion-label id="open-run-time-modal" slot="end">{{ job?.runTime ? getTime(job.runTime) : $t('Select run time') }}</ion-label>
+        <ion-label @click="setOpen(true)" slot="end">{{ job?.runTime ? getTime(job.runTime) : $t('Select run time') }}</ion-label>
```
Add a TODO that we are facing issues with trigger for modal.
We could directly set isModalOpen to true

adityasharma7 replied:
As checked, the issue is because we have JobConfiguration component twice. Use v-if for desktop only component

meet-aniket replied:
Sir, updated code.

adityasharma7 commented on src/components/JobConfiguration.vue:
```diff
@@ -10,13 +10,13 @@
         now commented the button and added a fallback string -->
         <!-- <ion-button id="open-run-time-modal" size="small" fill="outline" color="medium" v-show="!job?.runTime">{{ $t("Select run time") }}</ion-button> -->
-        <ion-modal trigger="open-run-time-modal">
+        <ion-modal :is-open="isModalOpen" @didDismiss="setOpen(false)">
```
We could directly set isModalOpen to false

meet-aniket replied:
Sir, updated code.

adityasharma7 commented on src/views/JobDetails.vue:
```diff
@@ -36,5 +37,57 @@ export default defineComponent({
+      }
+    }
+  },
+  mounted() {
```
I think we should check if this is the right lifecycle hook, we should not need any watcher in JobConfiguration component

meet-aniket committed:
Improved: code to handle the case when we are not able to open date-time picker on mobile screen (#24b1ket).

meet-aniket committed:
Improved: code to apply trigger property instead of using isOpen to open date-time picker (#24b1ket).

meet-aniket committed:
Improved: code to update reactive variable to computed property in jobConfiguration
component so that we can update our title as we are getting it from parent component (#24b1ket).

meet-aniket committed:
Improved: code to revert unwanted changes from code (#24b1ket).

meet-aniket committed:
Improved: code to update current job state on taking any action on particular job (#24b1ket).

meet-aniket committed:
Improved: code to update action name getCurrentJob -> updateCurrentJob (#24b1ket).

meet-aniket committed:
Improved: code to update the logic to update current job and removed unwanted prop property 'job' (#24b1ket).

adityasharma7 commented on src/store/modules/job/actions.ts:
```diff
@@ -529,5 +539,51 @@ const actions: ActionTree<JobState, RootState> = {
+      return payload?.job;
+    }
+
+    const currentJob = pendingJobs.find((job: any) => job.jobId === payload.jobId) ? pendingJobs.find((job: any) => job.jobId === payload.jobId) : cachedJobs[payload?.jobId];
```
We make currentJob let, assign it pendingJobs.find((job: any) => job.jobId === payload.jobId) and further it doesn't has value assign it cachedJobs[payload?.jobId]

meet-aniket replied:
Sir, updated code as per comments.

adityasharma7 commented on src/store/modules/job/actions.ts:
```diff
@@ -529,5 +539,51 @@ const actions: ActionTree<JobState, RootState> = {
+        }
+        commit(types.JOB_CURRENT_UPDATED, currentJob);
+
+        const enumIds = [] as any;
```
```suggestion
        const enumIds = resp.data.docs.map((item: any) => {
          return item.systemJobEnumId;
        })
```

meet-aniket replied:
Sir, updated code.

adityasharma7 commented on src/views/JobDetails.vue:
```diff
@@ -36,5 +37,57 @@ export default defineComponent({
+    viewJobConfiguration(job: any) {
+      const jobCategory = this.$route.params.category;
+      if(jobCategory !== 'pipeline') {
+        this.title = this.$route.params.title ? this.$route.params.title : this.jobTitles[job?.systemJobEnumId];
```
We could improve this code

meet-aniket commented on src/store/modules/job/actions.ts:
```diff
@@ -529,5 +530,46 @@ const actions: ActionTree<JobState, RootState> = {
     }
     return resp;
   },
+  currentJobUpdated({ commit }, payload){
```
Sir, updated code.

meet-aniket commented on src/store/modules/job/actions.ts:
```diff
@@ -529,5 +530,46 @@ const actions: ActionTree<JobState, RootState> = {
+  currentJobUpdated({ commit }, payload){
+    commit(types.JOB_CURRENT_UPDATED, payload);
+  },
+  async getCurrentJob({ commit, state, dispatch }, payload) {
```
Sir, updated the code as per your comments.

meet-aniket committed:
Improved: code to update check condition for current job action related to job
module and updated refectoring of response to extract enumIds from it (#24b1ket).

meet-aniket committed:
Improved: code to improve the logic to fetch ids and jobFreqTypes (#24b1ket).

meet-aniket committed:
Improved: code to update 'cancleJob' action to fetch and update current job (#24b1ket).

ymaheshwari1 commented on src/views/Orders.vue:
```diff
@@ -273,16 +276,22 @@ export default defineComponent({
       this.currentJobStatus = status
       this.freqType = id && this.jobFrequencyType[id]
-
+      
```
Remove unwanted change

meet-aniket replied:
Sir, updated code.

meet-aniket committed:
Improved: code to remove unwanted extra changes from code like:
-> extra spacing.
-> pushing status and type of job into route params (#24b1ket).

meet-aniket committed:
Improved: code to update payload to fetchJob in updateJob action
(#24b1ket).
